### PR TITLE
feat: UX improvement to address blockaid performance issue

### LINKED ISSUE #7521: [Bug]: PPOM - It takes too long to trigger the transaction modal with the feature enabled
### Describe the bug

**Problem**: It takes too long to trigger the transaction modal with the feature enabled. My observations with a Malicious ER20 Approval from the [test dapp](https://metamask.github.io/test-dapp/):

1. [Mobile]: With Blockaid enabled: it takes ~ 8 seconds to see the tx modal
2. [Mobile] With Blockaid disabled: it takes ~ <1 second to see the tx modal
3. [Extension] With Blockaid enabled: it takes ~ 1 second to see the tx popup

### Expected behavior

Which should be the threshold here? cc @bschorchit 

### Screenshots

1. [Mobile]: With Blockaid enabled

https://github.com/MetaMask/metamask-mobile/assets/54408225/3154ec2c-a8e1-4880-8bf2-32e56c408af5

2. [Mobile]: With Blockaid disabled:

https://github.com/MetaMask/metamask-mobile/assets/54408225/4e4d36a4-35d4-4c77-9dce-819964d721e0

3. [Extension] With Blockaid enabled

https://github.com/MetaMask/metamask-mobile/assets/54408225/3439f626-0a34-4a56-b26a-0723297c1196

### Steps to reproduce

1. Select Mainnet
2. Enable Blockaid
3. Go to the test dapp
4. Click Malicious Approval

### Error messages or log output

_No response_

### Version

Integrate PPOM branch  (behind feature flag) https://github.com/MetaMask/metamask-mobile/pull/7363

### Build type

None

### Device

Pixel 6

### Operating system

Android

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**
UX improvement for blockaid performance issue. Currently we intervene incoming transaction and blockaid UI rendering till request is validated. With the change validation is made an async background process so UI loading is not blocked for it.

## **Related issues**

Fixes: #7521

## **Manual testing steps**

1. Go to test dapp.
2. Trigger a transaction
3. For the first transaction which takes longer loader should be visible on transaction page

## **Screenshots/Recordings**
https://github.com/MetaMask/metamask-mobile/assets/2182307/75916125-cfc0-452f-9b0a-ec64f1dccbcf

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by jpuri** [comment]
Bitrise: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/218a29f5-2264-449d-8637-d6c0e0acbfe2

File: app/lib/ppom/ppom-util.ts [hunk 1]
```diff
@@ -28,18 +38,37 @@ const validateRequest = async (req: any) => {
...
-    return;
+  } finally {
+    if (!securityAlertResponse) {
+      securityAlertResponse = FailedResponse;
+    }
+    if (
+      req.method === 'eth_sendRawTransaction' ||
+      req.method === 'eth_sendTransaction'
+    ) {
+      store.dispatch(updateTransaction({ securityAlertResponse }));
```
Comment (segun) [inline app/lib/ppom/ppom-util.ts:62]: We're calling update transaction here and also calling `TransactionController.updateSecurityAlertResponse` in the next line. Why is this call not sufficient?

**Review by segun — DISMISSED** [review]
Approved with comment.

File: app/lib/ppom/ppom-util.ts (hunk 1 above)
Comment (jpuri) [reply app/lib/ppom/ppom-util.ts:62]: After the change in this PR that call will not be needed, result us asynchronously updated in `ppom-util`

File: app/actions/signatureRequest/index.js
Comment (blackdevelopa) [inline app/actions/signatureRequest/index.js:1]: looking good, but we should have new files be .ts

File: app/components/UI/BlockaidBanner/BlockaidBanner.tsx [hunk 2]
```diff
@@ -56,14 +84,50 @@ const BlockaidBanner = (bannerProps: BlockaidBannerProps) => {
...
     return null;
   }
 
+  if (!securityAlertResponse) {
+    return (
+      <View style={styles.marginedBanner}>
+        <BannerAlert
+          severity={BannerAlertSeverity.Warning}
+          title={strings('blockaid_banner.loading_title')}
+          startAccessory={<ActivityIndicator size="small" color="#F66A0A" />}
```
Comment (blackdevelopa) [inline app/components/UI/BlockaidBanner/BlockaidBanner.tsx:105]: I think we should import colour from theme. Might be wrong, but this looks like `colors.warning.default`

File: app/reducers/signatureRequest/index.js
Comment (blackdevelopa) [inline app/reducers/signatureRequest/index.js:1]: https://github.com/MetaMask/metamask-mobile/pull/7680/files#r1386609338

File: app/components/UI/BlockaidBanner/BlockaidBanner.tsx (hunk 2 above)
Comment (jpuri) [reply app/components/UI/BlockaidBanner/BlockaidBanner.tsx]: Done

File: app/actions/signatureRequest/index.js
Comment (jpuri) [reply app/actions/signatureRequest/index.js:1]: Done

**Review by blackdevelopa — DISMISSED** [review]

**Review by blackdevelopa — APPROVED** [review]

**Review by segun — APPROVED** [review]

**Pull request merged**